diff --git a/nixos_rebuild/__init__.py b/nixos_rebuild/__init__.py
index 78f3098..d46368c 100644
--- a/nixos_rebuild/__init__.py
+++ b/nixos_rebuild/__init__.py
@@ -367,6 +367,8 @@ def main() -> None:
     except (Exception, KeyboardInterrupt) as ex:
         if logger.isEnabledFor(logging.DEBUG):
             raise
+        elif isinstance(ex, KeyboardInterrupt):
+            sys.exit("\n\nInterrupted by user")
         else:
             sys.exit(str(ex))
 
